src/Apps/Artist/Routes/Overview/state.ts

```diff
@@ -71,15 +114,30 @@ export class FilterState extends Container<State> {
     }
   }
 
-  setPage(page, mediator) {
+  previousQueryString = ""
+  pushHistory() {
+    const currentQueryString = urlFragmentFromState(this.state)
+    // PriceRangeFilter's onAfterChange event fires twice; this ensures
+    //   we only push that history event once.
+    if (this.previousQueryString !== currentQueryString) {
```

Comment:
The comment pretty well describes it, but here's where we have to maintain the previous query string, so we can compare the current one. Without this check, we end up with duplicate history items when you interact with the Price Range slider. I _believe_ this is because the `rc-slider` dependency is unexpectedly firing an event twice, but I didn't chase it down much further than that. This felt like a reasonable workaround to me.

Reply:
This code, and the comment, makes total sense – thanks for the clarity! Does this appear to be a bug in `rc-slider`? Could we open an issue reporting it to the authors? Or, if one already exists, could we link to it from this comment?

Reply:
I am not totally sure. I _suspect_ that it is, but I haven't taken the time to isolate the actual issue. 

There are a couple issues in the rc-slider repo that are closed but have comments on them indicating they are not actually fixed ([466](https://github.com/react-component/slider/issues/466) and [361](https://github.com/react-component/slider/issues/361)). I suspect those are related but again...I need to invest more time to isolate the issue.